gammapy/spectrum/spectrum_analysis.py

```diff
@@ -406,94 +427,207 @@ def _check_binning(self):
         pass
 
 
-def run_spectrum_analysis_using_configfile(configfile):
-    """Wrapper function to run a 1D spectral analysis using a config file
+class SpectralFit(object):
+    """
+    Spectral Fit
 
     Parameters
     ----------
-    configfile : str
-        Config file in YAML format
-
-    Returns
-    -------
-    analysis : `~gammapy.spectrum.SpectrumAnalysis`
-        Spectrum analysis instance
+    pha : list of str, `~gammapy.extern.pathlib.Path`
+        List of PHA files to fit
     """
-    import yaml
-    log.info('Reading {}'.format(configfile))
-    with open(configfile) as fh:
-        config = yaml.safe_load(fh)
 
-    analysis = run_spectrum_analysis_using_config(config)
-    return analysis
+    def __init__(self, pha, bkg=None, arf=None, rmf=None):
+
+        self.pha = [make_path(f) for f in pha]
+        self._model = None
+        self._thres_lo = None
+        self._thres_hi = None
 
+    @classmethod
+    def from_config(cls, config):
 
-def run_spectrum_analysis_using_config(config):
-    """Wrapper function to run a 1D spectral analysis using a config dict
+        outdir = make_path(config['general']['outdir'])
+        pha_list = outdir.glob('pha_run*.pha')
+        return cls(pha_list)
+
+    @property
+    def model(self):
+        """
+        Spectral model to be fit
+        """
+        return self._model
+
+    @model.setter
+    def model(self, model, name=None):
+        """
+
+        Parameters
+        ----------
+        model : `~sherpa.models.ArithmeticModel`
+            Fit model
+        name : str
+            Name for Sherpa model instance, optional
+        """
+        import sherpa.models
+
+        name = 'default' if name is None else name
+
+        if isinstance(model, six.string_types):
+            if model == 'PL':
+                model = sherpa.models.PowLaw1D('powlaw1d.'+name)
+                model.gamma = 2
+                model.ref = 1e9
+                model.ampl = 1e-20
```

Comment:
This needs an else ValueError clause or it will fail later with an error that's hard to understand if the user mistypes "PowerLaw" instead of "PL", no?